src/extension.ts

```diff
@@ -33,24 +41,49 @@ class PSCompletionItem extends CompletionItem {
 }
 
 class PSCompletionProvider implements CompletionItemProvider {
-  constructor(private snippets: [Snippet]) {}
+  constructor(private snippets: Snippets) {}
 
   provideCompletionItems(): ProviderResult<CompletionItem[]> {
     return Object.keys(snippets).map(key => new PSCompletionItem(snippets[key]))
   }
 }
 
 export async function activate(context: ExtensionContext) {
-  let registerCompletionProvider = languages.registerCompletionItemProvider(
-    'javascript',
-    new PSCompletionProvider(snippets as [Snippet]),
-    '*'
-  )
+  async function workspaceFoldersChange(folders: WorkspaceFolder[]) {
```

Comment:
A bit miss-leading identifier, IMO. Does `registerSnippetProvider` expresses the intent better?